File: Lightened Dream/ExportForm.vb
Public Class ExportForm

  Private m_boolChecking As Boolean

  Private Sub ExportForm_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load

  End Sub

  Private Sub btnCancel_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnCancel.Click
    Close()
  End Sub

  Private Sub btnOk_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnOk.Click

  End Sub

  Private Sub chkSelectAll_CheckedChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles chkSelectAll.CheckedChanged
    m_boolChecking = True
    CheckNodes(trvMain.Nodes, chkSelectAll.Checked)
    m_boolChecking = False
  End Sub

  Private Sub CheckNodes(ByVal Nodes As TreeNodeCollection, ByVal Checked As Boolean)
    For Each trvNode As TreeNode In Nodes
      trvNode.Checked = Checked
      CheckNodes(trvNode.Nodes, Checked)
    Next
  End Sub

  Private Sub trvMain_AfterCheck(ByVal sender As System.Object, ByVal e As System.Windows.Forms.TreeViewEventArgs) Handles trvMain.AfterCheck
    If m_boolChecking Then Return

    CheckNodes(e.Node.Nodes, e.Node.Checked)
  End Sub
End Class

File: Lightened Dream/ExportForm.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Linq;
using System.Windows.Forms;
namespace LightenedDream
{
	public partial class ExportForm
	{


		private bool m_boolChecking;

		private void ExportForm_Load(System.Object sender, System.EventArgs e)
		{
		}

		private void btnCancel_Click(System.Object sender, System.EventArgs e)
		{
			Close();
		}


		private void btnOk_Click(System.Object sender, System.EventArgs e)
		{
		}

		private void chkSelectAll_CheckedChanged(System.Object sender, System.EventArgs e)
		{
			m_boolChecking = true;
			CheckNodes(trvMain.Nodes, chkSelectAll.Checked);
			m_boolChecking = false;
		}

		private void CheckNodes(TreeNodeCollection Nodes, bool Checked)
		{
			foreach (TreeNode trvNode in Nodes) {
				trvNode.Checked = Checked;
				CheckNodes(trvNode.Nodes, Checked);
			}
		}

		private void trvMain_AfterCheck(System.Object sender, System.Windows.Forms.TreeViewEventArgs e)
		{
			if (m_boolChecking)
				return;

			CheckNodes(e.Node.Nodes, e.Node.Checked);
		}
		public ExportForm()
		{
			Load += ExportForm_Load;
			InitializeComponent();
		}
	}
}
